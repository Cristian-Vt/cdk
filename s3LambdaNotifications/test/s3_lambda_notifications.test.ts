import { Template } from 'aws-cdk-lib/assertions';
import {App, Stack, aws_lambda} from "aws-cdk-lib";
import lambda = require('../lib/s3_lambda_notifications-stack');
// import { SynthUtils } from '@aws-cdk/assert';
// import'@aws-cdk/assert/jest'

// enhanced way
test('lambda_resource', () => {

  const app = new App();

  const s3LambdaTriggerStack  = new Stack(app, "S3LambdaTrigger")

  const lambda_resource  = new lambda.S3LambdaNotificationsStack(s3LambdaTriggerStack, 'Lambda');

  const template = Template.fromStack(lambda_resource)

  // using build in cdk library to test
  template.hasResource('AWS::Lambda::Function', {});

// using external library to test
  // expect(lambda_resource).toHaveResource('AWS::Lambda::Function', {
  //   MemorySize: 1024
  // })

});
  
// succint way
// test('lambda_resource', () => {

//   const stack = new Stack();
//   const lambda_resource  = new lambda.S3LambdaNotificationsStack(stack, 'Lambda');

//   expect(lambda_resource).toHaveResource('AWS::Lambda::Function', {
//     MemorySize: 1024
//   })
// });