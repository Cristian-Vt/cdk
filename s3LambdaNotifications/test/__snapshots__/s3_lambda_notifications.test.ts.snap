// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test stack to match snapshot 1`] = `Object {}`;
